### PR TITLE
Data Pipeline V2: Rename all Output implementations to end in Output

### LINKED ISSUE #937: Rename all `Output` implementations to end in `Output`
E.g. `Probabilities` to become `ProbabilitiesOutput`.
Follow-up to #927 for standardisation of names.
Need to deprecate old ones too

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

Fixes #937

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ethanwharris — COMMENTED** [review]
Looks like some classes are missing from here: https://github.com/PyTorchLightning/lightning-flash/blob/master/flash/core/classification.py

e.g. `Labels` should become either `LabelsOutput` or `LabelsClassificationOutput`

Mind renaming those ones too?

File: flash/core/classification.py [hunk 1]
```diff
@@ -339,3 +339,20 @@ def transform(
             filepath = sample[DataKeys.METADATA]["filepath"]
             return {"filepath": filepath, "predictions": fo_predictions}
         return fo_predictions
+
+
+class Labels(LabelsOutput):
+    def __init__(self):
```
Comment (ethanwharris) [inline flash/core/classification.py:345]:
```suggestion
    def __init__(self, labels: Optional[List[str]] = None, multi_label: bool = False, threshold: float = 0.5):
```

File: flash/core/classification.py [hunk 2]
```diff
@@ -339,3 +339,20 @@ def transform(
...
+        rank_zero_deprecation(
+            "`Labels` was deprecated in v0.6.0 and will be removed in v0.7.0." "Please use `LabelsOutput` instead."
+        )
+        super().__init__()
```
Comment (ethanwharris) [inline flash/core/classification.py:349]:
```suggestion
        super().__init__(labels=labels, multi_label=multi_label, threshold=threshold)
```

File: flash/core/classification.py [hunk 3]
```diff
@@ -339,3 +339,20 @@ def transform(
...
+
+
+class Probabilities(ProbabilitiesOutput):
+    def __init__(self):
```
Comment (ethanwharris) [inline flash/core/classification.py:353]:
```suggestion
    def __init__(self, multi_label: bool = False):
```

File: flash/core/classification.py [hunk 4]
```diff
@@ -339,3 +339,20 @@ def transform(
...
+        rank_zero_deprecation(
+            "`Probabilities` was deprecated in v0.6.0 and will be removed in v0.7.0."
+            "Please use `ProbabilitiesOutput` instead."
+        )
+        super().__init__()
```
Comment (ethanwharris) [inline flash/core/classification.py:358]:
```suggestion
        super().__init__(multi_label=multi_label)
```

File: flash/core/classification.py (hunk 1 above)
Comment (tchaton) [reply flash/core/classification.py]: Using *arg, **kwargs instead.

**Review by ethanwharris — APPROVED** [review]
Awesome, LGTM :smiley:

**Pull request merged**